src/tag/tag.scss

```diff
@@ -1,83 +1,88 @@
 // See LICENSE.md in the project root for license terms and full copyright notice.
 @import '../style/index';
-@import '../icon/index';
 @import '../text/mixins';
+@import '../button/index';
 
-@mixin iui-tag {
+@mixin iui-tag-base {
   @include iui-reset;
   @include iui-focus($offset: -2px, $thickness: 2px);
   user-select: all;
   text-transform: lowercase;
   display: inline-flex;
+}
+
+@mixin iui-tag-basic {
+  @include iui-tag-base;
+  outline-offset: 1px;
 
   &:not(a) {
     @include themed {
       color: t(iui-text-color-muted);
     }
   }
+}
 
-  &:not(.iui-basic) {
-    height: $iui-baseline * 3;
-    margin: $iui-component-offset 0;
-    border-radius: ($iui-baseline * 3) * 0.5;
-    box-sizing: border-box;
-    padding: 0 $iui-xxs;
-    align-items: center;
-    @media (prefers-reduced-motion: no-preference) {
-      transition: border $iui-speed-fast ease-out;
-    }
-    font-size: $iui-font-size;
-    text-transform: none;
-    cursor: default;
-    @include themed {
-      border: 1px solid rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
-      background-color: t(iui-color-background-1);
-      color: t(iui-text-color);
-    }
-
-    > .iui-label {
-      @include iui-text;
-      white-space: nowrap;
-      text-overflow: ellipsis;
-      overflow: hidden;
-      margin: 0 $iui-xs 0 $iui-s;
+@mixin iui-tag {
+  @include iui-tag-base;
+  height: $iui-baseline * 3;
+  margin: $iui-component-offset 0;
+  border-radius: ($iui-baseline * 3) * 0.5;
+  box-sizing: border-box;
+  padding: 0 $iui-xxs;
+  align-items: center;
+  @media (prefers-reduced-motion: no-preference) {
+    transition: border $iui-speed-fast ease-out;
+  }
+  font-size: $iui-font-size;
+  text-transform: none;
+  cursor: default;
+  @include themed {
+    border: 1px solid rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
+    background-color: t(iui-color-background-1);
+    color: t(iui-text-color);
+  }
 
-      &:only-child {
-        margin: 0 $iui-s;
-      }
+  &:hover {
+    @include themed {
+      border-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-1));
     }
+  }
+}
 
-    > .iui-button {
-      border-radius: 50%;
-    }
+@mixin iui-tag-label {
+  @include iui-text;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  overflow: hidden;
+  margin: 0 $iui-xs 0 $iui-s;
 
-    &:hover {
-      @include themed {
-        border-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-1));
-      }
-    }
+  &:only-child {
+    margin: 0 $iui-s;
   }
+}
 
-  &.iui-basic {
-    outline-offset: 1px;
-  }
+@mixin iui-tag-button {
+  @include iui-button;
+  @include iui-button-borderless;
+  @include iui-button-size($size: small, $styleType: borderless);
+  border-radius: 50%;
 }
 
 @mixin iui-tag-container {
   @include themed {
     color: t(iui-text-color-muted);
   }
 
-  > a.iui-basic {
+  > a.iui-tag-basic {
     margin: $iui-xxs;
   }
 
-  > .iui-tag:not(:last-child) {
+  > *:not(last-child) {
```

Comment:
What if you add gap with `@support` check so in the future we could easily find it and remove with IE11 deprecation?

Reply:
I have tried that, but then lists with `iui-truncate` applied breaks...
![image](https://user-images.githubusercontent.com/81580355/161524381-ee34ea00-826a-49df-b782-77bf649e67be.png)
(notice ellipsis not having margin...)

Should I then add margins back if `iui-truncate` applied?
I couldnt find any info about `-webkit-box` display we use.. Mozila page has its link broken, so not sure.

Reply:
Doesn't make sense to apply margins with truncate, then it is ok.